components/persistor/lib/knex/db.ts

```diff
@@ -1243,22 +1243,22 @@ module.exports = function (PersistObjectTemplate) {
 
                     let toDeletePromiseArr = [];
 
-                    // create our `delete functions` to be run later.
-                    // also put them in one place => toDeletePromiseArr.
-                    for (const key of persistorTransaction.remoteObjects) {
-                        toDeletePromiseArr.push(async () => {
-                            try {
-                                await remoteDocService.deleteDocument(key, this.bucketName);
-                            } catch (e) {
-                                (logger || this.logger).error({
-                                    component: 'persistor',
-                                    module: 'api',
-                                    activity: 'end',
-                                    error: e},
-                                    'unable to rollback remote document with key:' + key + ' and bucket: ', this.bucketName);
-                            }
-                        });
-                    }
+                    persistorTransaction.remoteObjects.forEach((versionId: string, key: string) => {
+                        toDeletePromiseArr.push(
+                            remoteDocService.deleteDocument(key, this.bucketName, versionId)
```

Comment:
Does this rollback to previous version of the document on update

1. Add Document Version **1**
2. commit successfully
3. Update to Document Version **2**
4. commit fails, rollback

Is Document 1 restored?

Reply:
It's only deleting a specific version, so as long as you have versioning on then you will still retain `version 1`